lib/screens/pets/add-pet/add_pet_form.dart

```diff
@@ -128,10 +130,16 @@ class _AddPetFormState extends State<AddPetForm> {
     petObject.type = typeController.text;
     petObject.biography = biographyController.text;
     petObject.age = int.parse(ageController.text);
-    _storageService.uploadPhoto(_image).then((pictureUrl) {
-      petObject.pictureUrl = pictureUrl;
-      //widget.addPetHandler(petObject);
-    });
+
     widget.addPetHandler(petObject);
+
+    if (_image != null) {
+      _storageService.uploadPhoto(_image).then((pictureUrl) {
+        petObject.pictureUrl = pictureUrl;
+
+        _petsService
+            .updatePet(petObject);
+      });
+    }
   }
```

Comment:
Since `addPetHandler` is async, it is possible for `updatePet` to be called before the service was added.
Please fix this.